File: demo/src/domain/Ticket.ts
import type { Product } from "./Product";

export interface TicketLine {
  product: Product;
  quantity: number;
  lineTotal: number;
  excludingTax: number;
  vatRate: number;
}

export interface Ticket {
  lines: TicketLine[];
  totalHt: number;
  vat5: number;
  vat20: number;
  totalTtc: number;
  discountAmount: number;
  fidPEarned: number;
  fidBalance: number;
}

File: engine/src/main/java/checkoutengine/Ticket.java
package checkoutengine;

import java.util.List;

import checkoutengine.VAT.VatBreakdownLine;

public record Ticket (
    List<TicketLine> lines, 
    double totalHt,
    List<VatBreakdownLine> vatBreakdown,
    double totalTtc,
    //added for fidelity points in v5
    double discountAmount,
    int fidPEarned,
    int fidBalance,
    String discountName
)
{
    
}
